glusterd2/commands/volumes/volume-option.go

```diff
@@ -62,45 +62,66 @@ func optionSetValidate(c transaction.TxnCtx) error {
 }
 
 type txnOpType uint8
-type volumeOpType uint8
 
 const (
 	txnDo txnOpType = iota
 	txnUndo
-	volumeSet volumeOpType = iota
-	volumeReset
 )
 
 func xlatorActionDoSet(c transaction.TxnCtx) error {
-	return xlatorAction(c, txnDo, volumeSet)
+	return xlatorAction(c, txnDo, xlator.VolumeSet)
 }
 
 func xlatorActionUndoSet(c transaction.TxnCtx) error {
-	return xlatorAction(c, txnUndo, volumeSet)
+	return xlatorAction(c, txnUndo, xlator.VolumeSet)
 }
 
 func xlatorActionDoReset(c transaction.TxnCtx) error {
-	return xlatorAction(c, txnDo, volumeReset)
+	return xlatorAction(c, txnDo, xlator.VolumeReset)
 }
 
 func xlatorActionUndoReset(c transaction.TxnCtx) error {
-	return xlatorAction(c, txnUndo, volumeReset)
+	return xlatorAction(c, txnUndo, xlator.VolumeReset)
+}
+
+func xlatorActionDoVolumeStart(c transaction.TxnCtx) error {
+	return xlatorAction(c, txnDo, xlator.VolumeStart)
+}
+
+func xlatorActionUndoVolumeStart(c transaction.TxnCtx) error {
+	return xlatorAction(c, txnUndo, xlator.VolumeStart)
+}
+
+func xlatorActionDoVolumeStop(c transaction.TxnCtx) error {
+	return xlatorAction(c, txnDo, xlator.VolumeStop)
+}
+
+func xlatorActionUndoVolumeStop(c transaction.TxnCtx) error {
+	return xlatorAction(c, txnUndo, xlator.VolumeStop)
 }
 
 // This function can be reused when volume reset operation is implemented.
 // However, volume reset can be also be treated logically as volume set but
 // with the value set to default value.
-func xlatorAction(c transaction.TxnCtx, txnOp txnOpType, volOp volumeOpType) error {
+func xlatorAction(c transaction.TxnCtx, txnOp txnOpType, volOp xlator.VolumeOpType) error {
 	reqOptions := make(map[string]string)
-	if volOp == volumeSet {
+	var fn func(*volume.Volinfo, string, string, xlator.VolumeOpType, log.FieldLogger) error
+	var volinfo volume.Volinfo
+
```

Comment:
nit: Declare variables right before they are unmarshalled. Let's focus on better readability and leave the pre-declaration optimisation to the compiler.

```go
var volinfo volume.Volinfo
if err := c.Get("volinfo", &volinfo); err != nil {
....
```